### DISCUSSION TITLE
Autoblock issues with eigh

### DISCUSSION THREAD

**Discussion by ProThicc** [jcmgray/quimb#223] [General]
I noticed a potential issue with autoblock method for eigh. The autoblock method gives a wrong set of eigenvalues and eigenvectors for the same Hamiltonian.

I tested this out for a lot of other Hermitian Hamiltonians and there seems to be no issue at all.
This issue only pops up with this particular hamiltonian I defined below and num_qubits >= 11

Here is the code highlighting the issue

```python

import quimb as qu
num_qubits = 11
esp = qu.up() @ qu.up().H
lsp = qu.down() @ qu.down().H
X = qu.pauli('x')
H = sum([qu.pkron((esp & X & lsp) + (lsp & X & esp) ,dims = [2]*num_qubits,inds = [(i-1)%num_qubits, i%num_qubits, (i+1)%num_qubits]) for i in range(num_qubits)])

lk1,vk1 = qu.eigh(H, autoblock=False, sort = True, isherm = True, return_vecs = True)
lk2,vk2 = qu.eigh(H, autoblock=True, sort = True, isherm = True, return_vecs = True)


lk1
array([-7.02667418, -6.74204451, -6.74204451, ...,  6.74204451,
        6.74204451,  7.02667418])

lk2
array([-6.98262848, -6.45741483, -5.8667071 , ...,  6.66307307,
        6.70920627,  6.98262848])
``` 
The Hamiltonian is hermitian, but as you can notice the output of the eigenvalues is different. I am unable to figure out what the issue might be for this.

**Comment by jcmgray** [comment]
Yes puzzling, I get the same thing - will take a look into it.

**Reply by ProThicc** [reply]
I tried to check if the sectors formed using this method might be non-hermitian for some reason for the above special case by putting up a check in the autoblock.py file. It seems that, it is not the issue.

**Reply by jcmgray** [reply]
The issue is that the `compute_blocks` algorithm is not fully working for finding all disconnected subgraphs (it appears to need a further merging step), should be possible to fix relatively easily.

**Reply by jcmgray** [reply]
This should be fixed by https://github.com/jcmgray/quimb/commit/a2c6d27535baaa7e781892e227b3bd532e6d506d, let me know if not!

**Reply by ProThicc** [reply]
Thanks @jcmgray It works perfectly!
